phpBB/phpbb/db/tools/doctrine.php

```diff
@@ -384,6 +418,31 @@ public function sql_truncate_table(string $table_name): void
 		}
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function add_prefix(string $name, string $prefix): string
+	{
+		return strpos($prefix, '_', -1) ? $prefix . $name : $prefix . '_' . $name;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function remove_prefix(string $name, string $prefix = ''): string
+	{
+		$prefix = strpos($prefix, '_', -1) ? $prefix : $prefix . '_';
+		return $prefix && self::is_prefixed($name, $prefix) ? substr($name, strlen($prefix)) : $name;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function is_prefixed(string $name, string $prefix): bool
+	{
+		return strpos($name, $prefix) === 0;
```

Comment:
Not sure whether it makes sense to have this a separate method but use https://www.php.net/manual/en/function.str-starts-with.php